sql/plan/innerjoin.go

```diff
@@ -4,15 +4,46 @@ import (
 	"io"
 	"os"
 	"reflect"
+	"runtime"
+	"strconv"
 
 	opentracing "github.com/opentracing/opentracing-go"
+	"github.com/pbnjay/memory"
+	"github.com/sirupsen/logrus"
 	"gopkg.in/src-d/go-mysql-server.v0/sql"
 )
 
-const experimentalInMemoryJoinKey = "EXPERIMENTAL_IN_MEMORY_JOIN"
-const inMemoryJoinSessionVar = "inmemory_joins"
+const (
+	experimentalInMemoryJoinKey = "EXPERIMENTAL_IN_MEMORY_JOIN"
+	maxMemoryJoinKey            = "MAX_MEMORY_INNER_JOIN"
+	inMemoryJoinSessionVar      = "inmemory_joins"
+	memoryThresholdSessionVar   = "max_memory_joins"
+)
+
+var (
+	useInMemoryJoins = os.Getenv(experimentalInMemoryJoinKey) != ""
+	// Half of the total physical memory available on the OS (ignoring the
+	// memory used by other processes).
+	defaultMemoryThreshold = memory.TotalMemory() / 2
```

Comment:
This is maybe too much? should reduce this to use just like 15-20% of the memory available? Take into account that the object's cache is only using 512mb as default param.

Reply:
Ok, I can change it to total memory / 5 or memory / 6